docs/user-guides/plugin/plugin-tests.md

```diff
@@ -138,6 +138,54 @@ helm upgrade <Release name> <chart-path>
 helm test <Release name>
 ```
 
+## Plugin Testing with dependencies
+
+Some plugins require other plugins to be installed in the cluster for their tests to run successfully. To support this, each plugin can declare required dependencies using a `test-dependencies.yaml` file.
+
+> [!NOTE]  
+> The `test-dependencies.yaml` file is **required** if other plugins need to be installed in the cluster before running the tests in the GitHub Actions workflow during a Pull Request for the plugin.
+
+
+### How It Works
+- Each plugin can optionally include a `test-dependencies.yaml` file in the plugin’s root directory (e.g., `alerts/test-dependencies.yaml`).
+- The file defines a list of plugin names that should be installed **before** testing begins.
+
+### Example `test-dependencies.yaml`
+```yaml
+dependencies:
+  - cert-manager
+```
+
+In this example, the `alert` plugin declares that `cert-manager` plugin must be installed first.
+
+### Automation during Pull Requests
+The GitHub Actions workflow automatically:
+
+1. Detects plugins that are changed in the pull request.
+2. Parses the `test-dependencies.yaml` for each changed plugin if present.
+3. Installs listed dependencies in order
+4. Proceeds with helm chart linting and testing
+
+## Declaring Test Values for CI Testing
+
+Each plugin should provide a `test-values.yaml` file to supply necessary values during testing. This file should be placed inside the `<PLUGIN-NAME>/ci` directory.
```

Comment:
Checked the linked PR but maybe you could add some context that the `test-values.yaml` of the declared dependency are used to deploy the required Plugin's chart. I was not sure if the `test-values.yaml` of the Plugin that has changed need to contain all the configuration for the Plugin it depends on as well.